qualtran/bloqs/qft/qft_text_book.py

```diff
@@ -11,22 +11,44 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
+from collections import defaultdict
 from functools import cached_property
-from typing import Iterator
+from typing import Iterator, Set
 
 import attrs
 import cirq
 from numpy.typing import NDArray
+from sympy import Expr
 
-from qualtran import GateWithRegisters, QUInt, Signature
+from qualtran import bloq_example, BloqDocSpec, GateWithRegisters, QUInt, Signature
+from qualtran._infra.bloq import Bloq
+from qualtran.bloqs.basic_gates.hadamard import Hadamard
+from qualtran.bloqs.basic_gates.swap import TwoBitSwap
 from qualtran.bloqs.rotations.phase_gradient import PhaseGradientUnitary
+from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
 from qualtran.symbolics import SymbolicInt
+from qualtran.symbolics.types import is_symbolic
 
 
 @attrs.frozen
 class QFTTextBook(GateWithRegisters):
-    r"""Textbook version of QFT using $\frac{N (N + 1)}{2}$ controlled rotations.
+    r"""Quantum Fourier Transform as presented in Chapter 5.1 of Nielsen and
+    Chuang
 
+    Performs the standard QFT on a register of `bitsize` qubits utilizing
+    `bitsize` Hadamards and `bitsize` * (`bitsize` - 1) / 2 controlled Z
+    rotations, along with a reversal of qubit ordering specified via
+    `with_reverse` which defaults to `True`. `bitsize` can be provided numerically or symbolically. More specific QFT implementations
+    can be found:
```

Comment:
wrap this text so it looks normal in the python files

Reply:
fixed